### PR TITLE
E2-1815: download files directly from URL instead of API call

### PR DESCRIPTION
**Pull request by e7mac**
As per https://leanplum.atlassian.net/browse/E2-1815 (Proposal 1), we will be sending the `files` key with the URL for each file. If this exists, we want the SDK to use the URL directly to get these files.

### CODE REVIEW

File: Leanplum-SDK/Classes/Managers/Networking/LPFileTransferManager.m [hunk 1]
```diff
@@ -274,6 +274,8 @@ - (void)downloadFile:(NSString *)path onResponse:(LPNetworkResponseBlock)respons
         id<LPNetworkOperationProtocol> op;
         if ([path hasPrefix:@"http://"] || [path hasPrefix:@"https://"]) {
             op = [self.engine operationWithURLString:path];
+        } else if ([self.filenameToURL valueForKey:path]) {
+            op = [self.engine operationWithURLString:[self.filenameToURL valueForKey:path]];
```
Comment (gg4race) [inline Leanplum-SDK/Classes/Managers/Networking/LPFileTransferManager.m]: should we still check whether the value here has prefix http or https? other than that lgtm

**Comment by gg4race** [comment]
did we look into adding some tests?

File: Leanplum-SDK/Classes/Internal/Leanplum.m [hunk 2]
```diff
@@ -2372,6 +2375,8 @@ + (void)forceContentUpdate:(LeanplumVariablesChangedBlock)block
         NSDictionary *regions = response[LP_KEY_REGIONS];
         NSDictionary *variantDebugInfo = [self parseVariantDebugInfoFromResponse:response];
         [[LPVarCache sharedCache] setVariantDebugInfo:variantDebugInfo];
+        NSDictionary *filenameToURL = [self parseFileURLsFromResponse:response];
```
Comment (hrishileanplum) [inline Leanplum-SDK/Classes/Internal/Leanplum.m]: This is a list of files. Can we please rename this to something like "filenameToUrls"

File: Leanplum-SDK/Classes/Internal/Leanplum.m [hunk 3]
```diff
@@ -2372,6 +2375,8 @@ + (void)forceContentUpdate:(LeanplumVariablesChangedBlock)block
...
+        [LPFileTransferManager sharedInstance].filenameToURL = filenameToURL;
```
Comment (hrishileanplum) [inline Leanplum-SDK/Classes/Internal/Leanplum.m]: same here.

File: Leanplum-SDK/Classes/Internal/Leanplum.m (hunk 2 above)
Comment (hrishileanplum) [reply Leanplum-SDK/Classes/Internal/Leanplum.m]: Also, 1 check , if "filenameToUrls" is nil, then not to assign it to [LPFileTransferManager sharedInstance].filenameToURL

File: Leanplum-SDK/Classes/Managers/Networking/LPFileTransferManager.m (hunk 1 above)
Comment (e7mac) [reply Leanplum-SDK/Classes/Managers/Networking/LPFileTransferManager.m]: that's unnecessary. The filenameToURL should always be giving us a URL

File: Leanplum-SDK/Classes/Internal/Leanplum.m (hunk 2 above)
Comment (e7mac) [reply Leanplum-SDK/Classes/Internal/Leanplum.m]: We don't want that check. If for whatever reason, we decide to stop sending the dictionary from the backend, we don't want the last non-nil value to be retained forever.

**Pull request merged**